api/turing/config/config.go

```diff
@@ -32,85 +40,100 @@ func (qty *Quantity) Decode(value string) (err error) {
 	return nil
 }
 
+// MarshalJSON implements the json.Marshaller interface so that we will get the expected
+// JSON string representation when marshalling a quantity object.
+func (qty *Quantity) MarshalJSON() ([]byte, error) {
+	q := resource.Quantity(*qty)
+	return q.MarshalJSON()
+}
+
 // Config is used to parse and store the environment configs
 type Config struct {
-	Port                int                  `envconfig:"turing_port" default:"8080"`
-	AllowedOrigins      []string             `split_words:"true" default:"*"`
-	AuthConfig          *AuthorizationConfig `envconfig:"authorization" validate:"required"`
-	DbConfig            *DatabaseConfig      `envconfig:"turing_database"`
-	DeployConfig        *DeploymentConfig    `envconfig:"deployment"`
-	RouterDefaults      *RouterDefaults      `envconfig:"turing_router"`
-	Sentry              sentry.Config        `split_words:"false" envconfig:"SENTRY"`
-	VaultConfig         *VaultConfig         `envconfig:"vault"`
-	TuringEncryptionKey string               `split_words:"true" required:"true"`
-	AlertConfig         *AlertConfig         `envconfig:"alert"`
-	MLPConfig           *MLPConfig
-	TuringUIConfig      *TuringUIConfig `envconfig:"turing_ui"`
+	Port                int `validate:"required"`
+	AllowedOrigins      []string
+	AuthConfig          *AuthorizationConfig
+	DbConfig            *DatabaseConfig   `validate:"required"`
+	DeployConfig        *DeploymentConfig `validate:"required"`
+	RouterDefaults      *RouterDefaults   `validate:"required"`
+	Sentry              sentry.Config
+	VaultConfig         *VaultConfig `validate:"required"`
+	TuringEncryptionKey string       `validate:"required"`
+	AlertConfig         *AlertConfig
+	MLPConfig           *MLPConfig `validate:"required"`
+	TuringUIConfig      *TuringUIConfig
 	// SwaggerFile specifies the file path containing OpenAPI spec. This file will be used to configure
 	// OpenAPI validation middleware, which validates HTTP requests against the spec.
-	SwaggerFile string `envconfig:"swagger_file" default:"swagger.yaml"`
+	SwaggerFile string
 	// Experiment specifies the JSON configuration to set up experiment managers and runners.
 	//
 	// The configuration follows the following format to support different experiment engines
 	// and to allow custom configuration for different engines:
 	// { "<experiment_engine>": <experiment_engine_config>, ... }
 	//
 	// For example:
-	// { "experiment_engine_a": `{"client": "foo"}`, "experiment_engine_b": `{"apikey": 12}` }
-	Experiment map[string]string `envconfig:"experiment"`
+	// { "experiment_engine_a": {"client": "foo"}, "experiment_engine_b": {"apikey": 12} }
+	Experiment map[string]interface{}
```

Comment:
Updated from `map[string]string` to `map[string]interface{}` so that in the config file, this can be specified like so:
```
Experiment:
  expA:
    key: val
```
rather than
```
Experiment:
   expA: {"key":"val"}
```